ui/src/components/Publication/Creation/MainText.tsx

```diff
@@ -47,68 +161,81 @@ const MainText: React.FC = (): React.ReactElement | null => {
         }
     }, [user, publicationId]);
 
-    const addReferences = React.useCallback(async () => {
-        try {
-            const referenceField: React.RefObject<any> = addReferencesRef;
-
-            // Pattern using regex groups to retrieve reference text and to determine whether a string contains DOI or URI
-            const pattern =
-                /(?<TEXT>.+?(?=http))((?<DOI>((((http|https):\/\/)(([-a-zA-Z0-9_]{1,265}([^\s]+)))))(10{1}\.([^\n]+)))|(?<URL>((((http|https):\/\/)(([-a-zA-Z0-9_]{1,265}([^\n]+)))))))|(?<TEXTONLY>.+?(?=$))/g;
-
-            const referenceFieldValue = referenceField.current.value;
-            const matches = referenceFieldValue ? referenceFieldValue.matchAll(pattern) : undefined;
-
-            // Iterate through matches
-            if (matches) {
-                let matchArray: Interfaces.Reference[] = [];
-                for (const match of matches) {
-                    let location: string | null;
-                    let type: Interfaces.ReferenceType;
-                    let text: string;
-
-                    if (match.groups.DOI) {
-                        location = match.groups.DOI;
-                        type = 'DOI';
-                        text = match.groups.TEXT;
-                    } else if (match.groups.URL) {
-                        location = match.groups.URL;
-                        type = 'URL';
-                        text = match.groups.TEXT;
-                    } else {
-                        type = 'TEXT';
-                        text = match.groups.TEXTONLY;
-                        location = null;
+    const addReferences = React.useCallback(
+        async (editorContent: string) => {
+            try {
+                const paragraphsArray = editorContent.match(/<p>(.*?)<\/p>/g) || [];
+
+                if (!paragraphsArray.length) {
+                    return;
+                }
+
+                for (let i = 0; i < paragraphsArray.length; i++) {
+                    const currentParagraph = paragraphsArray[i].trim();
+                    const paragraphElement = new DOMParser()
+                        .parseFromString(currentParagraph, 'text/html')
+                        .querySelector('p');
+                    const textContent = paragraphElement?.textContent?.trim();
+
+                    if (!textContent) {
+                        return;
```

Comment:
Instead of returning here, if this becomes `continue;` instead, it keeps the order the allows for the double lines without inserting spaces, or at least it did for me 😄

Reply:
yeah, I can't remember exactly why I added this return right here. The UI is updating accordingly now. Good catch! 😄